### PR TITLE
ITE soc/riscv-ite/it8xxx2: move config to soc from board

### PR DESCRIPTION
**Pull request by RuibinChang**
Make config conditional and move to soc Kconfig.defconfig.series
from it8xxx2_evb_defconfig.

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>

### CODE REVIEW

**Comment by RuibinChang** [comment]
As @MaureenHelm suggestion in tachometer PR (https://github.com/zephyrproject-rtos/zephyr/pull/39993).

NOTE: After tachometer PR is merged, this PR should be rebased and solved conflict then could be merged .

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series [hunk 1]
```diff
@@ -16,19 +16,57 @@ config ITE_IT8XXX2_INTC
...
+config I2C_ITE_IT8XXX2
+	default y
+	depends on I2C
+
+config ADC_ITE_IT8XXX2
+	default y
+	depends on ADC
+
+config SOC_FLASH_ITE_IT8XXX2
+	default y
```
Comment (MaureenHelm) [inline soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series:64]: `depends on FLASH`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for following up on this, it looks good. The next step is to add a second commit that removes the following from `boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig`:
- CONFIG_WATCHDOG=y
- CONFIG_KSCAN=y
- CONFIG_I2C=y
- CONFIG_PWM=y

The net result after the two commits will be that these drivers are no longer enabled in hello_world and the right SoC-specific drivers are enabled in applications that need them.

**Review by keith-zephyr — APPROVED** [review]
LGTM with Maureen's requested change.

**Review by brockus-zephyr — APPROVED** [review]
LGTM with Maureen's request.

File: soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series (hunk 1 above)
Comment (RuibinChang) [reply soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series:64]: Done

Comment (RuibinChang) [reply soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series:64]: So if I want to build the it8xxx2_evb with tachometer driver, then I should add the CONFIG_SENSOR=y in samples/boards/it8xxx2_evb/prj.conf (if the path exists) , right?

Comment (MaureenHelm) [reply soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series:64]: Yes

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**